services/external_sources/kaggle.py

```diff
@@ -37,21 +37,25 @@ def kaggle_search(query, owner, limit=5, prev=0):
     logger.debug(f"Searching kaggle for query: {query}")
     res = []
     try:
-        command = f"kaggle datasets list --file-type csv -s {query} --csv --max-size 5000000"
-        if prev % 20 == 0:
-            command += f" --page {math.floor(prev/20)+1}"
+        if query == "":
+            command = f"kaggle datasets list --file-type csv --csv --max-size 5000000"
+        else:
+            command = f"kaggle datasets list --file-type csv -s {query} --csv --max-size 5000000"
+
+        command += f" --page {math.floor(prev/20)+1}"
         output = subprocess.check_output(command, shell=True, text=True)
         # if the first three letters are not 'ref' return an error
         if output[0:3] != "ref":
             logger.error(f"Error in kaggle_search - ref is not found, maybe there is a kaggle update: {output}")
-            return "error"
+            return []
 
         df = pd.read_csv(io.StringIO(output))
         # for row in df...
         for i in range(len(df)):
-            if i < prev:
+            index = i + math.floor(prev/20) * 20
```

Comment:
When adding magic functions like this, explain what they are for, and why they are there. values like `20` should not be magic numbers but a constant preferably with an explanation of why their value is X

Also, why 20 and why is this function in place?